src/prefect/tasks/dremio/dremio.py

```diff
@@ -84,14 +127,80 @@ def run(self, user: str, password: str, host: str, port: int, query: str) -> Dic
         """
         if not query:
             raise ValueError("A query string must be provided")
+        scheme = "grpc+tcp"
+        connection_args = {}
+
+        if tls:
+            # Connect to the server endpoint with an encrypted TLS connection.
+            print("[INFO] Enabling TLS connection")
+            scheme = "grpc+tls"
+            if certs:
+                print("[INFO] Trusted certificates provided")
+                # TLS certificates are provided in a list of connection arguments.
+                with open(certs, "rb") as root_certs:
+                    connection_args["tls_root_certs"] = root_certs.read()
+            else:
+                print(
+                    """[ERROR] Trusted certificates must be provided to establish a TLS connection."""
+                )
+                sys.exit()
+        else:
+            print(
+                "[WARNING] You are not using a secure conncetion. Consider setting tls=True"
+            )
+
+        # Two WLM settings can be provided upon initial authneitcation
+        # with the Dremio Server Flight Endpoint:
+        # - routing-tag
+        # - routing queue
+        initial_options = flight.FlightCallOptions(
+            headers=[
+                (b"routing-tag", b"test-routing-tag"),
+                (b"routing-queue", b"Low Cost User Queries"),
+            ]
+        )
+        client_auth_middleware = DremioClientAuthMiddlewareFactory()
+        client = flight.FlightClient(
+            "{}://{}:{}".format(scheme, hostname, flightport),
+            middleware=[client_auth_middleware],
+            **connection_args,
+        )
 
-        client = flight.FlightClient(f"grpc+tcp://{host}:{port}")
-        client.authenticate(HttpDremioClientAuthHandler(user, password))
-
-        info = client.get_flight_info(
-            flight.FlightDescriptor.for_command(query + "--arrow flight")
+        # Authenticate with the server endpoint.
+        bearer_token = client.authenticate_basic_token(
+            username, password, initial_options
+        )
+        print("[INFO] Authentication was successful. Token is valid for 30 hours.")
```

Comment:
Can you use `self.logger` instead so these will show up in the UI?

Reply:
A bunch of this seems DEBUG level

Reply:
Thanks for your feedback @madkinsz. I used `self.logger` where appropriate and removed redundant print statements.